addons/sale_timesheet/views/project_task_views.xml

```diff
@@ -131,13 +131,13 @@
                         column_invisible="not parent.allow_billable"
                         readonly="readonly_timesheet"
                         context="{'with_remaining_hours': True, 'with_price_unit': True}" options="{'no_create': True, 'no_open': True}"
-                        domain="[('is_service', '=', True), ('order_partner_id', 'child_of', parent.partner_id), ('is_expense', '=', False), ('state', '=', 'sale'), ('is_downpayment', '=', False)]"
+                        domain="[('is_service', '=', True), ('order_partner_id.commercial_partner_id.id', 'parent_of', parent.partner_id), ('is_expense', '=', False), ('state', '=', 'sale'), ('is_downpayment', '=', False)]"
```

Comment:
in `order_partner_id.commercial_partner_id.id`, the `.id` is really needed? 🧐 
Same comment in the domain defined below.

Reply:
I'm not sure why, but it seems that using .id is necessary. While order_partner_id.commercial_partner_id looks sufficient, It might require the explicit id for consistency.

Reply:
TBH, I don't think it is needed and if `order_partner_id.commercial_partner_id` is sufficient then I would prefer keeping it simple.

Can you please remove `.id` in all your domain?

Thanks in advance!

Reply:
it is not working without `.id`

Reply:
Why? 🤔 

I will investigate because I am a bit surprised it is not equivalent.

Reply:
@mjvi-odoo indeed the behavior is different, I got the explanation in another PR.

https://github.com/odoo/odoo/pull/182945#discussion_r1796833544